src/iceberg/type_fwd.h

```diff
@@ -81,4 +81,36 @@ class TimestampTzType;
 class Type;
 class UuidType;
 
+/// \brief Error types for iceberg.
+/// TODO: add more and sort them based on some rules.
+enum class ErrorKind {
+  kNoSuchNamespace,
+  kAlreadyExists,
+  kNoSuchTable,
+  kCommitStateUnknown,
+};
+
+struct Namespace;
```

Comment:
Since we are putting these forward declarations here, should we change the name type_fwd.h to something like spec_fwd.h?

Reply:
`type_fwd.h` is a convention used by arrow-cpp to hold forward declarations and enums in a single header file. It might be a coincident that it starts with the data types...

Reply:
Yeah, it's not "Iceberg data type forward declarations", it's "C++ data type forward declarations". Open to renaming it (though STL at least uses the `fwd` convention too with `iosfwd`)